client/src/components/Calendar.jsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import moment from 'moment';
+import styled from 'styled-components';
+
+
+class Calendar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dateObject: moment(),
+      leftMonth: moment(),
+      rightMonth: moment().add(1, 'months'),
+      leftDays: null,
+      rightDays: null,
+      dateSelected: null
+    };
+  }
+
+  previousMonth() {
+    let left = moment(this.state.leftMonth).subtract(1, 'months');
+    let right = moment(this.state.rightMonth).subtract(1, 'months');
+    this.setState({
+      leftMonth: left,
+      rightMonth: right
+    });
+  }
+    
+  nextMonth() {
+    let left = moment(this.state.leftMonth).add(1, 'months');
+    let right = moment(this.state.rightMonth).add(1, 'months');
+    this.setState({
+      leftMonth: left,
+      rightMonth: right
+    });
+  }
+
+  firstDayOfMonth(leftOrRight) {
+    let firstDay = moment(leftOrRight).startOf('month').format('d');
+    return firstDay;
+  }
+
+  render() {
+    const eachDayOfWeek = moment.weekdaysShort().map(day => (
+      <DayOfMonth key={day}>
+        {day.slice(0, -1)}
+      </DayOfMonth>
+    ));
+
+    let leftFiller = [];
+    for (let i = 0; i < this.firstDayOfMonth(this.state.leftMonth); i += 1) {
+      leftFiller.push(
+        <CalendarDay style={{ border: 'none' }}></CalendarDay>
+      );
+    }
+
+    let rightFiller = [];
+    for (let i = 0; i < this.firstDayOfMonth(this.state.rightMonth); i += 1) {
+      rightFiller.push(
+        <CalendarDay style={{ border: 'none' }}></CalendarDay>
+      );
+    }
```

Comment:
I guess `let rightFiller = this.firstDayOfMonth(this.state.rightMonth).map(...)` could make it easier.  Same for `leftFiller`.